### ISSUE TITLE
CS0165 is not reported for uninitialized DateTimeOffsets in .NET Standard 2.0

### ISSUE THREAD

**Issue by hankovich** [dotnet/roslyn#43951]
**Version Used**: 
Visual C# Compiler version 3.4.1-beta4-19614-01 (https://github.com/dotnet/roslyn/commit/165046097562cfe65b09c2e9a9d8f7cd88526f2c)

**Steps to Reproduce**:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>

</Project>
```

```csharp
using System;

namespace CompanyName
{
    public static class Program
    {
        public static void Main()
        {
            DateTimeOffset now;
            Console.WriteLine(now);
        }
    }
}
```

**Expected Behavior**:
Program.cs(10,31,10,34): error CS0165: Use of unassigned local variable 'now'

**Actual Behavior**:
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========

Error is reported when other types are used instead of DateTimeOffset (I tested with DateTime, DateTimeOffset?, int, DayOfWeek, etc.).
Error is also reported for netcoreapp3.1, netstandard2.1 and net471.

But for DateTimeOffset & netstandard2.0 error is not reported. Moreover, if we will reference this project from some project with an entry point and call ```CompanyName.Program.Main()```, `1/1/0001 12:00:00 AM +00:00` will be printed.

**Comment by CyrusNajmabadi** [comment]
The likely issue is htat you're using reference assemblies that don't contain members in structs.  Because the struct has no members, it is thought to be always assigned (since there's no data in it to actually assign).

**Comment by sharwell** [comment]
Duplicate of #29319
